### ISSUE TITLE
Cannot unfollow suspended users / no unfollow button for banned users

### ISSUE THREAD

**Issue by socdoneleft** [mastodon/mastodon#22466]
### Steps to reproduce the problem

1. Follow an (unsuspended) user
2. That user is now suspended (eg, https://mastodon.social/@NotJustBikes@mstdn.social )
3. Go to your profile, then "following" (eg, https://mastodon.social/@socdoneleft/following )
4. The suspended user has no visible "unfollow" button.

### Expected behaviour

It should be possible to unfollow suspended users.

### Actual behaviour

It is not possible to unfollow suspended users.

### Detailed description

I followed YouTuber NotJustBikes' first Mastodon account on mstdn.social. That account has since been suspended (and moved to https://social.notjustbikes.com/@notjustbikes ):

<img width="386" alt="image" src="https://user-images.githubusercontent.com/49317004/208323136-04d3d977-4823-4ecb-9635-5af68e6e1a11.png">

I would like to unfollow this suspended account, in order to give me a better idea of how many active accounts I follow. However, in the "following" page on my profile, no such unfollow button appears:

<img width="592" alt="image" src="https://user-images.githubusercontent.com/49317004/208323111-940633ee-0653-4840-8b9e-66c48c9a699f.png">

The only similar issue that appeared for "unfollow" and "suspened" is #20961 ( https://github.com/mastodon/mastodon/issues/20961 ), which appears to be the result of a "blocked remote instance". However, I don't believe that mastodon.social and mstdn.social have blocked each other -- the top post in mstdn.social's Explore tab is one from mastodon.social!

<img width="591" alt="image" src="https://user-images.githubusercontent.com/49317004/208323309-2c6d84b9-9b44-4c7b-8c62-996289745f34.png">

**TLDR: I would like to be able to unfollow suspended accounts.**

### Specifications

Mastodon: v4.0.2 (mastodon.social)
Chrome: Version 108.0.5359.98 (Official Build) (arm64)

**Comment by socdoneleft** [comment]
Addendum: It seems to be possible to unfollow users via the "Follows and followers" menu (Preferences > Follows and followers):
<img width="800" alt="image" src="https://user-images.githubusercontent.com/49317004/208349206-b535e3cd-d730-4f15-9b95-ae0beb0fe54a.png">

**Comment by cachelina** [comment]
Hey! I'm willing to implement this.
